src/clustering/administration/main/command_line.cc

```diff
@@ -645,6 +650,244 @@ service_address_ports_t get_service_address_ports(const std::map<std::string, op
         port_offset);
 }
 
+bool load_tls_key_and_cert(SSL_CTX *tls_ctx, const std::string &key_file, const std::string &cert_file) {
+    if(SSL_CTX_use_PrivateKey_file(tls_ctx, key_file.c_str(), SSL_FILETYPE_PEM) <= 0) {
+        ERR_print_errors_fp(stderr);
+        return false;      
+    }
+
+    if(SSL_CTX_use_certificate_file(tls_ctx, cert_file.c_str(), SSL_FILETYPE_PEM) <= 0) {
+        ERR_print_errors_fp(stderr);
+        return false;      
+    }
+
+    if(1 != SSL_CTX_check_private_key(tls_ctx)) {
+        logNTC("private key and certificate do not match");
```

Comment:
(also capitalization and punctuation: "The private key and certificate do not match.")